.github/workflows/test-release.yml

```diff
@@ -0,0 +1,38 @@
+name: Test Release
+
+on:
+  pull_request:
+  push:
+    branches:
+      - staging
+
+jobs:
+  build-n-publish:
+    name: Build and publish Python 🐍 distributions 📦 to TestPyPI
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@master
+
+    - name: Set up Python 3.9
+      uses: actions/setup-python@v1
+      with:
+        python-version: '3.10'
+
+    - name: Ensure latest setuptools
+      run: |
+          python -m pip install --upgrade pip setuptools wheel
+
+    - name: Build wheel and source distribution
+      run: |
+          python setup.py sdist bdist_wheel
+
+    - name: Publish distribution 📦 to Test PyPI
+      uses: pypa/gh-action-pypi-publish@master
```

Comment:
earlier this used to be done only with tagged commits. now all commits on staging branch will get pushed? any particular reason for this change?

Reply:
my bad, you're correct.